operator/crdgen/main.go

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2021-2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/gravitational/trace"
+
+	gogodesc "github.com/gogo/protobuf/protoc-gen-gogo/descriptor"
+	"github.com/gogo/protobuf/protoc-gen-gogo/generator"
+	gogoplugin "github.com/gogo/protobuf/protoc-gen-gogo/plugin"
+	"github.com/gogo/protobuf/vanity/command"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+)
+
+func main() {
```

Comment:
How challenging would it be to introduce a test that runs this against a super simple proto, and compares it against some generated files in `testdata/` ?

Reply:
Please take a look at https://github.com/gravitational/teleport/pull/13331/commits/ebe20548bbad221f6ebfcd76e1538484fff9b2f8